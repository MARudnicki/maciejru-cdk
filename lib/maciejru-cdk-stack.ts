// import * as sqs from 'aws-cdk-lib/aws-sqs';

import {Construct, Stack, StackProps} from "@aws-cdk/core";

export class MaciejruCdkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here

    // example resource
    // const queue = new sqs.Queue(this, 'MaciejruCdkQueue', {
    //   visibilityTimeout: cdk.Duration.seconds(300)
    // });
  }
}
