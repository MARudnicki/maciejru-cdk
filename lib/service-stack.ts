import {Construct, Stack, StackProps} from "@aws-cdk/core";
import {CfnParametersCode, Code, Function, Runtime} from "@aws-cdk/aws-lambda";
import {HttpApi, HttpMethod} from "@aws-cdk/aws-apigatewayv2";
import {HttpProxyIntegration, LambdaProxyIntegration} from "@aws-cdk/aws-apigatewayv2-integrations";

export class ServiceStack extends Stack {

    public readonly serviceCode: CfnParametersCode

    constructor(scope: Construct, id: string, props?: StackProps) {
        super(scope, id, props);

        this.serviceCode = Code.fromCfnParameters()

        const lambda: Function = new Function(this, "ServiceLambda", {
            runtime: Runtime.NODEJS_14_X,
            handler: 'src/lambda.handler',
            code: this.serviceCode,
            functionName: 'ServiceLambda'
        })

        const innerApiGateway: HttpApi = new HttpApi(this, 'InnerApiGateway', {
            defaultIntegration: new LambdaProxyIntegration({
                handler: lambda
            }),
            apiName: 'InnerApiGateway'
        })

        const outerApiGateway = new HttpApi(this, 'outerApiGateway')
        outerApiGateway.addRoutes({
            path: "/",
            methods: [HttpMethod.GET],
            integration: new HttpProxyIntegration({
                url: innerApiGateway.url || "",
            })
        })

        outerApiGateway.addRoutes({
            path: "/test",
            methods: [HttpMethod.GET],
            integration: new HttpProxyIntegration({
                url: "https://google.com"
            })
        })

    }
}
