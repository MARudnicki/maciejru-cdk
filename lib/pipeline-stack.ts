import {Construct, SecretValue, Stack, StackProps} from "@aws-cdk/core";
import {Artifact, IStage, Pipeline} from "@aws-cdk/aws-codepipeline";
import {
    CloudFormationCreateUpdateStackAction,
    CodeBuildAction,
    GitHubSourceAction
} from "@aws-cdk/aws-codepipeline-actions";
import {BuildSpec, LinuxBuildImage, PipelineProject} from "@aws-cdk/aws-codebuild";
import {ServiceStack} from "./service-stack";
import {BillingStack} from "./biling-stack";

export class PipelineStack extends Stack {

    private readonly pipeline: Pipeline;
    private readonly cdkBuildOutput: Artifact;
    private readonly lambdaBuildOutput: Artifact;


    constructor(scope: Construct, id: string, props: StackProps) {
        super(scope, id, props);

        this.pipeline = new Pipeline(this, 'Pipeline', {
            pipelineName: 'MyPipeline',
            crossAccountKeys: false,
            restartExecutionOnUpdate: true
        })

        const cdkSourceOutput = new Artifact('CdkSourceOutput')
        const lambdaSourceOutput = new Artifact('LambdaSourceOutput')

        this.pipeline.addStage({
            stageName: 'MySource',
            actions: [
                new GitHubSourceAction({
                    owner: 'MARudnicki',
                    repo: 'maciejru-cdk',
                    branch: 'master',
                    actionName: 'Pipeline_Source',
                    oauthToken: SecretValue.secretsManager('github-token'),
                    output: cdkSourceOutput
                }),
                new GitHubSourceAction({
                    owner: 'MARudnicki',
                    repo: 'express-lambda',
                    branch: 'master',
                    actionName: 'Lambda_Source',
                    oauthToken: SecretValue.secretsManager('github-token'),
                    output: lambdaSourceOutput
                })

            ]
        })

        this.cdkBuildOutput = new Artifact("CdkBuildOutput")
        this.lambdaBuildOutput = new Artifact("LambdaBuildOutput")

        this.pipeline.addStage({

            stageName: 'MyBuild',
            actions: [
                new CodeBuildAction({
                    actionName: 'CDK_Build',
                    input: cdkSourceOutput,
                    outputs: [this.cdkBuildOutput],
                    project: new PipelineProject(this, 'CdkBuildProject', {
                        environment: {
                            buildImage: LinuxBuildImage.STANDARD_5_0
                        },
                        buildSpec: BuildSpec.fromSourceFilename('build-spec/cdk-build-spec.yml')
                    })
                }),
                new CodeBuildAction({
                    actionName: 'Service_Build',
                    input: lambdaSourceOutput,
                    outputs: [this.lambdaBuildOutput],
                    project: new PipelineProject(this, 'LambdaBuildProject', {
                        environment: {
                            buildImage: LinuxBuildImage.STANDARD_5_0
                        },
                        buildSpec: BuildSpec.fromSourceFilename('build-spec/lambda-build-spec.yml')
                    })
                })
            ]

        })

        this.pipeline.addStage({
            stageName: 'MyPipelineUpdate',
            actions: [
                new CloudFormationCreateUpdateStackAction({
                    actionName: 'Pipeline_Update',
                    stackName: "MaciejruCdkStack",
                    templatePath: this.cdkBuildOutput.atPath("MaciejruCdkStack.template.json"),
                    adminPermissions: true
                })
            ]
        })
    }

    public addServiceStage(serviceStack: ServiceStack, stageName: string): IStage {
        return this.pipeline.addStage({
            stageName: stageName,
            actions: [
                new CloudFormationCreateUpdateStackAction({
                    actionName: 'Service_Update',
                    stackName: serviceStack.stackName,
                    templatePath: this.cdkBuildOutput.atPath(`${serviceStack.stackName}.template.json`),
                    adminPermissions: true,
                    parameterOverrides: {
                        ...serviceStack.serviceCode.assign(
                            this.lambdaBuildOutput.s3Location
                        )
                    },
                    extraInputs: [this.lambdaBuildOutput]
                })
            ]
        })
    }

    public addBillingStackToStage(bilingStack: BillingStack, stage: IStage) {
        stage.addAction(new CloudFormationCreateUpdateStackAction({
            actionName: 'Biling_Update',
            stackName: bilingStack.stackName,
            templatePath: this.cdkBuildOutput.atPath(
                `{bilingStack.stackName}.template.json`
            ),
            adminPermissions: true
        }))
    }
}
