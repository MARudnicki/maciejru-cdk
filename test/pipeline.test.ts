import {App} from "@aws-cdk/core";
import {ServiceStack} from "../lib/service-stack";
import {PipelineStack} from "../lib/pipeline-stack";
import {arrayWith, haveResourceLike, objectLike} from "@aws-cdk/assert";

test("Adding service stage", () => {
    ///Given
    const app = new App()
    const serviceStack = new ServiceStack(app, "ServiceStack")
    const pipelineStack = new PipelineStack(app, "PipelineStack", {})

    //When
    pipelineStack.addServiceStage(serviceStack, "Test")

    //Then
    expect(pipelineStack).toBe(
        haveResourceLike("AWS::CodePipeline::Pipeline")
    );
});

// test("Adding biling stacke", () => {
//     ///Given
//     const app = new App()
//     const serviceStack = new ServiceStack(app, "ServiceStack")
//     const pipelineStack = new PipelineStack(app, "PipelineStack", {})
//
//     //When
//     pipelineStack.addServiceStage(serviceStack, "Test")
//
//     //Then
//     expect(pipelineStack).toBe(
//         haveResourceLike("AWS::CodePipeline::Pipeline")
//     );
// });
