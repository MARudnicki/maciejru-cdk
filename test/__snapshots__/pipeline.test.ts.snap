// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline Stack 1`] = `
Object {
  "Resources": Object {
    "CdkBuildProject9382C38D": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "CODEPIPELINE",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "CdkBuildProjectRoleE0B6FEB0",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": "build-spec/cdk-build-spec.yml",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "CdkBuildProjectRoleDefaultPolicy3C7ECB00": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "CdkBuildProject9382C38D",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "CdkBuildProject9382C38D",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "CdkBuildProject9382C38D",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucket22248F97",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucket22248F97",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CdkBuildProjectRoleDefaultPolicy3C7ECB00",
        "Roles": Array [
          Object {
            "Ref": "CdkBuildProjectRoleE0B6FEB0",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CdkBuildProjectRoleE0B6FEB0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineArtifactsBucket22248F97": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "PipelineC660917D": Object {
      "DependsOn": Array [
        "PipelineRoleDefaultPolicyC7A05455",
        "PipelineRoleD68726F7",
      ],
      "Properties": Object {
        "ArtifactStore": Object {
          "Location": Object {
            "Ref": "PipelineArtifactsBucket22248F97",
          },
          "Type": "S3",
        },
        "Name": "MyPipeline",
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "PipelineRoleD68726F7",
            "Arn",
          ],
        },
        "Stages": Array [
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": Object {
                  "Branch": "master",
                  "OAuthToken": "{{resolve:secretsmanager:github-token:SecretString:::}}",
                  "Owner": "MARudnicki",
                  "PollForSourceChanges": false,
                  "Repo": "maciejru-cdk",
                },
                "Name": "Pipeline_Source",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "MySourceOutput",
                  },
                ],
                "RunOrder": 1,
              },
            ],
            "Name": "MySource",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ProjectName": Object {
                    "Ref": "CdkBuildProject9382C38D",
                  },
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "MySourceOutput",
                  },
                ],
                "Name": "CDK_Build",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "CdkBuildOutput",
                  },
                ],
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PipelineMyBuildCDKBuildCodePipelineActionRole8AF9C128",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "MyBuild",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ActionMode": "CREATE_UPDATE",
                  "Capabilities": "CAPABILITY_NAMED_IAM",
                  "RoleArn": Object {
                    "Fn::GetAtt": Array [
                      "PipelineMyPipelineUpdatePipelineUpdateRole57375C1F",
                      "Arn",
                    ],
                  },
                  "StackName": "MaciejruCdkStack",
                  "TemplatePath": "CdkBuildOutput::MaciejruCdkStack.template.json",
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "CdkBuildOutput",
                  },
                ],
                "Name": "Pipeline_Update",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PipelineMyPipelineUpdatePipelineUpdateCodePipelineActionRoleC1E38516",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "MyPipelineUpdate",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "PipelineMyBuildCDKBuildCodePipelineActionRole8AF9C128": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineMyBuildCDKBuildCodePipelineActionRoleDefaultPolicy6DE91FAB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "CdkBuildProject9382C38D",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineMyBuildCDKBuildCodePipelineActionRoleDefaultPolicy6DE91FAB",
        "Roles": Array [
          Object {
            "Ref": "PipelineMyBuildCDKBuildCodePipelineActionRole8AF9C128",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineMyPipelineUpdatePipelineUpdateCodePipelineActionRoleC1E38516": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineMyPipelineUpdatePipelineUpdateCodePipelineActionRoleDefaultPolicy6F30CEC2": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "iam:PassRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PipelineMyPipelineUpdatePipelineUpdateRole57375C1F",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucket22248F97",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucket22248F97",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "cloudformation:CreateStack",
                "cloudformation:DescribeStack*",
                "cloudformation:GetStackPolicy",
                "cloudformation:GetTemplate*",
                "cloudformation:SetStackPolicy",
                "cloudformation:UpdateStack",
                "cloudformation:ValidateTemplate",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":cloudformation:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":stack/MaciejruCdkStack/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineMyPipelineUpdatePipelineUpdateCodePipelineActionRoleDefaultPolicy6F30CEC2",
        "Roles": Array [
          Object {
            "Ref": "PipelineMyPipelineUpdatePipelineUpdateCodePipelineActionRoleC1E38516",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineMyPipelineUpdatePipelineUpdateRole57375C1F": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cloudformation.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineMyPipelineUpdatePipelineUpdateRoleDefaultPolicyF27F93AB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucket22248F97",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucket22248F97",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "*",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineMyPipelineUpdatePipelineUpdateRoleDefaultPolicyF27F93AB",
        "Roles": Array [
          Object {
            "Ref": "PipelineMyPipelineUpdatePipelineUpdateRole57375C1F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineMySourcePipelineSourceWebhookResource52E101F5": Object {
      "Properties": Object {
        "Authentication": "GITHUB_HMAC",
        "AuthenticationConfiguration": Object {
          "SecretToken": "{{resolve:secretsmanager:github-token:SecretString:::}}",
        },
        "Filters": Array [
          Object {
            "JsonPath": "$.ref",
            "MatchEquals": "refs/heads/{Branch}",
          },
        ],
        "RegisterWithThirdParty": true,
        "TargetAction": "Pipeline_Source",
        "TargetPipeline": Object {
          "Ref": "PipelineC660917D",
        },
        "TargetPipelineVersion": 1,
      },
      "Type": "AWS::CodePipeline::Webhook",
    },
    "PipelineRoleD68726F7": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineRoleDefaultPolicyC7A05455": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucket22248F97",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucket22248F97",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PipelineMyBuildCDKBuildCodePipelineActionRole8AF9C128",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PipelineMyPipelineUpdatePipelineUpdateCodePipelineActionRoleC1E38516",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineRoleDefaultPolicyC7A05455",
        "Roles": Array [
          Object {
            "Ref": "PipelineRoleD68726F7",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
